### PR TITLE
ReaderZooming: Deal with some more fallout of the new zoom modes

### PR DESCRIPTION
**Pull request by NiLuJe**
* Namely, ensure zoom_mode is consistent with genus & type *both ways*. (I only dealt with the "no zoom_mode" case in my original fixup).
Because documents with settings dating back from before the new zoom modes had "old" zoom_mode settings mixed with "new" genus/type defaults that didn't agree with each other.
It lead to super-confusing ConfigDialog behavior, because ConfigDialog was in fact not reflecting the reality.
(As the source of truth is actually `zoom_mode`).

* There was a snafu in manual mode, because of the extremely weird way prefixes are handled by Configurable/ReaderConfig/DocSettings/ConfigDialog.
So, make sure we only have a *single* zoom_factor, and that it's updated and saved properly under the right name everywhere.

Fixes inconsistencies between first swapping to manual mode, and what the ConfigDialog said/did (because again: possibly a lie), vs., re-opening the same document, which would magically use *different* settings, closer to what was expected (but still broken because of the prefix mismatch and a disagreement on defaults between the two variants).

Fallout from #6885

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7728)

### CODE REVIEW

**Comment by NiLuJe** [comment]
TL;DR: Keeping the old `zoom_mode` hybrid on top of the new `genus/type` combos has lead to incredibly messy messes.

**Comment by NiLuJe** [comment]
Ping @jperon to double-check I didn't mangle this too much ;).

File: frontend/apps/reader/modules/readerzooming.lua [hunk 1]
```diff
@@ -317,6 +334,7 @@ function ReaderZooming:onDefineZoom(btn, when_applied_callback)
     elseif zoom_mode == "manual" then
         if btn == "manual" then
             config.zoom_factor = self:getNumberOf("columns")
+            print("Manual set zoom_factor to", config.zoom_factor)
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerzooming.lua]: Print?

File: frontend/apps/reader/modules/readerzooming.lua [hunk 2]
```diff
@@ -346,9 +364,12 @@ function ReaderZooming:onDefineZoom(btn, when_applied_callback)
                 settings.zoom_overlap_h,
                 ("%.2f"):format(self:getNumberOf("rows", settings.zoom_overlap_v)),
                 settings.zoom_overlap_v,
-                ("%.2f"):format(self:getNumberOf("columns"))),
+                ("%.2f"):format(config.zoom_factor)),
             dismiss_callback = when_applied_callback,
         })
+    else
+        -- Refresh ConfigDialog, because we've just poked at its innards...
+        print("ReaderZooming mangled ConfigDialog ;)")
```
Comment (Frenzie) [inline frontend/apps/reader/modules/readerzooming.lua]: Ditto

File: frontend/ui/widget/configdialog.lua [hunk 3]
```diff
@@ -978,6 +978,7 @@ function ConfigDialog:onConfigEvents(option_events, arg_index)
 end
 
 function ConfigDialog:onConfigChoose(values, name, event, args, events, position, hide_on_apply)
+    print("ConfigDialog:onConfigChoose")
```
Comment (Frenzie) [inline frontend/ui/widget/configdialog.lua]: *ahem*

File: frontend/ui/widget/configdialog.lua [hunk 4]
```diff
@@ -1020,6 +1021,7 @@ function ConfigDialog:onConfigChoose(values, name, event, args, events, position
         -- need to update and repaint the whole config panel, as other
         -- toggles may have their state (enabled/disabled) modified
         -- after this toggle update.
+        print("ConfigDialog:onConfigChoose -> update")
```
Comment (Frenzie) [inline frontend/ui/widget/configdialog.lua]: And another :-P

**Comment by NiLuJe** [comment]
@Frenzie: You're looking at individual commits instead of the whole (this one is a "squash on merge" affair ;)).

**Review by poire-z — APPROVED** [review]
(I don't really want to dig and understand the logic :/ but nothing jumps out.)

**Review by Frenzie — APPROVED** [review]
Right, carry on then.

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 2 of 4 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/7728)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @NiLuJe)
